src/libserialize/tests/json.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(rustc::default_hash_types)]
```

Comment:
Could we switch these to `rustc::internal` like we do for `libserialize/lib.rs`?